src/html-imports.js

```diff
@@ -694,7 +685,28 @@
     document.addEventListener('load', onLoadingDone, true /* useCapture */ );
     document.addEventListener('error', onLoadingDone, true /* useCapture */ );
   } else {
-    new Importer();
+    // Override baseURI so that imported elements' baseURI can be used seemlessly
+    // on native or polyfilled html-imports.
+    // NOTE: a <link rel=import> will have `link.baseURI === link.href`, as the link
+    // itself is used as the `import` document.
+    const native_baseURI = Object.getOwnPropertyDescriptor(Node.prototype, 'baseURI');
```

Comment:
Tried in safari 8 and safari 6, in both it's possible to patch `Element.prototype.baseURI` 👌